source/includes/code-examples/InsertOneAsync.cs

```diff
@@ -0,0 +1,95 @@
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+using MongoDB.Bson.Serialization.Conventions;
+using MongoDB.Driver;
+using static System.Console;
+
+namespace CsharpExamples.UsageExamples;
+
+public class InsertOneAsync
+{
+    private static IMongoCollection<Restaurant> _restaurantsCollection;
+    private static string _mongoConnectionString = "<Your MongoDB URI>";
+
+    public static void Main(string[] args)
+    {
+        Setup();
+
+        WriteLine("Inserting a document...");
+        InsertOneRestaurant();
+
+        // find and print newly inserted document
+        var document = _restaurantsCollection.Find(Builders<Restaurant>.Filter
+            .Eq("name", "Mongo's Pizza")).First();
+
+        WriteLine("Document Inserted: " + document.ToBsonDocument());
+
+        Cleanup();
+
+    }
+
+    private static async void InsertOneRestaurant()
+    {
+        // delete sample document if already exists
+        Cleanup();
+
+        // start-insert-one-async
+        Restaurant newRestaurant = new Restaurant
```

Comment:
issue: I thought this could be simplified to just `Restaurant newRestaurant = new {`

Reply:
It can be changed to `Restaurant newRestaurant = new() { .... }`. Changed to that, let me know if that is better